File: 110.BalancedBinaryTree/clywin123/isBalanced.c
int max(int a,int b){
	if(a>b)
		return a;
	return b;
}
int height(struct TreeNode* root){
	if(root ==NULL)
		return 0;
	else
		return 1+max(height(root->left),height(root->right));
}

bool isBalanced(struct TreeNode* root) {
	int lv,rv,diff;
	if(root==NULL)
		return true;
	else{
		lv = height(root->left);
		rv = height(root->right);
		diff = lv-rv;
		if(diff<0)
			diff *= (-1);
		if(diff<=1){
			return isBalanced(root->left)&isBalanced(root->right);
		}
		else
			return false;
	}
}

File: 110.BalancedBinaryTree/wyrj/isBalanced.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isBalanced = function(root) {
    const f = (n) => {
        if (!n) return {b: true, h: 0};
        const l = f(n.left);
        const r = f(n.right);
        return {b: l.b && r.b && Math.abs(l.h - r.h) <= 1, h: Math.max(l.h, r.h) + 1};
    }
    return f(root).b;
};
